packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -222,14 +225,80 @@ const duplicateNameError = async (context: HookContext<LocationService>) => {
   }
 }
 
+/* BEFORE FIND HOOKS */
+/**
+ * @function rejectEmptyQuery
+ * @param context
+ * @description This hook function ensure that no empty queries are present
+ */
+const rejectEmptyQuery = async (context: HookContext<LocationService>) => {
+  const keys: string[] = Object.keys(context.params.query || {})
+  if (keys.length == 0) throw new BadRequest(`Query can't have empty params object`)
+}
+/**
+ * @function restrictProtectedQuery
+ * @param context
+ * @description This hook function restricts allowed query params for scoped user's calls
+ * from authorized sections (admin, dashboard, studio)
+ */
+const restrictProtectedQuery = async (context: HookContext<LocationService>) => {
+  keepQuery(
+    'id',
+    'name',
+    'slugifiedName',
+    '$like',
+    '$limit',
+    '$sort',
+    '$select',
+    '$skip',
+    'sceneId',
+    'projectId'
+  )(context)
+}
+/**
+ * @function restrictPublicQuery
+ * @param context
+ * @description This hook function restricts allowed query params for public calls
+ * from the viewer section by only allowing the slugifiedName parameter, any other
+ * parameter will be discarded from the query protecting from any malicious query injection.
+ *
+ * This function is discarding all params but slugifiedName instead of throwing a BadRequest if other
+ * parameters are present, having as goal to avoid providing any feedback to malicious users
+ * that could allow them to figure out unforeseen and unsafe patterns or combinations in order to
+ * successfully perform an injection attack.
+ */
+const restrictPublicQuery = async (context: HookContext<LocationService>) => {
+  keepQuery('slugifiedName')(context)
+}
+
 export default {
   around: {
     all: [schemaHooks.resolveExternal(locationExternalResolver), schemaHooks.resolveResult(locationResolver)]
   },
 
   before: {
     all: [schemaHooks.validateQuery(locationQueryValidator), schemaHooks.resolveQuery(locationQueryResolver)],
-    find: [discardQuery('action'), discardQuery('studio'), sortByLocationSetting],
+    find: [
+      // Empty query object in findLocation is not allowed
+      rejectEmptyQuery,
+      // Persist query to avoid hook collitions or malformed knex query instances
+      persistQuery,
+      // Clean up query of non db relevant parameters
+      discardQuery('action'),
+      // On external calls. evaluate action property presence and restrict queries according
+      iff(
+        isProvider('external'),
+        iffElse(
+          hasAction,
+          [
+            iff(isAction('admin'), verifyScope('location', 'read')).else(verifyScope('editor', 'write')),
```

Comment:
```suggestion
            iffElse(isAction('admin'), verifyScope('location', 'read'), verifyScope('editor', 'write')),
```

Just a humble request, since we use iffElse all across our codebase instead of iff().else(). Can you make this change.

Reply:
@hanzlamateen not sure why, but when iffElse is deep nested, intellisense is complaining while the exact same implementation at a top level isn't complaining, see the following:

![image](https://github.com/user-attachments/assets/a9e6a6cb-fc61-4b77-bc79-fca5bcac69bb)

that's not even the reason I decided to use the iff().else() approach, I just believe that it is unnecessary to allocate in memory 2 arrays each containing a single hookFunction, but I don't mind to address your change request it's just that for some reason intellisense is complaining even though these are supposed to be equivalent implementations.

thoughts?